internal/command/platform/status.go

```diff
@@ -2,31 +2,135 @@ package platform
+	"github.com/superfly/flyctl/internal/flag"
+	"github.com/superfly/flyctl/internal/logger"
+	"github.com/superfly/flyctl/internal/render"
 	"github.com/superfly/flyctl/iostreams"
 )
 
 func newStatus() (cmd *cobra.Command) {
 	const (
-		long = `Show current Fly platform status in a browser
+		long = `Show current Fly platform status in a browser or via json with the json flag
 `
-		short = "Show current platform status"
+		short = "Show current platform status with an optional filter for maintenance or incidents in json mode (eg. incidents, maintenance)"
 	)
 
-	cmd = command.New("status", short, long, runStatus)
+	cmd = command.New("status [kind](optional)", short, long, runStatus)
+	cmd.Args = cobra.MaximumNArgs(1)
+	return
+}
+
+type Page struct {
+	ID        string    `json:"id,omitempty"`
+	Name      string    `json:"name,omitempty"`
+	Url       string    `json:"url,omitempty"`
+	UpdatedAt time.Time `json:"updated_at,omitempty"`
+}
 
-	cmd.Args = cobra.NoArgs
+type Status struct {
+	Description string `json:"description,omitempty"`
+	Indicator   string `json:"indicator,omitempty"`
+}
 
-	return
+type incidentUpdates struct {
+	Body      string    `json:"body,omitempty"`
+	Status    string    `json:"status,omitempty"`
+	CreatedAt time.Time `json:"created_at,omitempty"`
+	UpdatedAt time.Time `json:"updated_at,omitempty"`
+}
+
+type Incident struct {
+	ID              string          `json:"id,omitempty"`
+	Name            string          `json:"name,omitempty"`
+	Status          string          `json:"status,omitempty"`
+	CreatedAt       time.Time       `json:"port,omitempty"`
+	Impact          string          `json:"impact,omitempty"`
+	MonitoringAt    time.Time       `json:"monitoring_at,omitempty"`
+	PageID          string          `json:"page_id,omitempty"`
+	ResolvedAt      time.Time       `json:"resolved_at,omitempty"`
+	UpdatedAt       time.Time       `json:"updated_at,omitempty"`
+	IncidentUpdates incidentUpdates `json:"incident_updates,omitempty"`
+}
+
+type StatusPage struct {
+	Status Status
+	Page   Page
+}
+type Client struct {
+	baseUrl    *url.URL
+	httpClient *http.Client
+}
+
+func createClient(ctx context.Context, rawUrl string) (*Client, error) {
+	cleanedURL, _ := url.Parse(rawUrl)
+	logger := logger.MaybeFromContext(ctx)
+	httpClient, err := api.NewHTTPClient(logger, http.DefaultTransport)
+	if err != nil {
+		return nil, fmt.Errorf("can't setup HTTP client to %s: %w", rawUrl, err)
+	}
+
+	return &Client{
+		baseUrl:    cleanedURL,
+		httpClient: httpClient,
+	}, nil
 }
 
 func runStatus(ctx context.Context) error {
-	const url = "https://status.fly.io/"
+	const url = "https://status.fly.io"
+	var (
+		cfg               = config.FromContext(ctx)
+		getStatusEndpoint string
+		getStatusKind     = flag.FirstArg(ctx)
+	)
+
+	switch getStatusKind {
+	case "incidents":
+		getStatusEndpoint = "api/v2/incidents/unresolved.json"
+	case "maintenance":
+		getStatusEndpoint = "api/v2/scheduled-maintenances/active.json"
```

Comment:
what do you think if it accepts `status` or empty string and return an error otherwise?

Reply:
Do you mean for the kind part of the command so `fly platform status status` or `fly platform status ` and those including maintenance and incidents?

Reply:
Yes. Although `fly platform status status` reads weird

Reply:
The whole point is that `fly platform status unknown-word` should return an error, not the status.

Reply:
ah ok gotcha